close_corporation now closes corporations

roseundy opened the pull request:
Fixes #7527 

There is a chance this will invalidate a few games.

tobymao commented on lib/engine/game/g_21_moon/game.rb:
```diff
@@ -920,6 +920,12 @@ def sellable_bundles(player, corporation)
           @round.active_step.sellable_bundles(player, corporation)
         end
 
+        def close_corporation(corporation, quiet: false)
```
why don't you just fix the base class?

roseundy replied:
I don't know what that will break. I assume there was a reason it didn't have it. I'll investigate.

roseundy replied:
See new commit.

roseundy commented:
OK, I've changed the base close_corporation method to add a corporation.close! call
I've removed the now redundant corporation.close! calls from several games and removed now redundant close_corporation overrides from a few games.
